rs/sns/root/src/lib.rs

```diff
@@ -704,12 +705,76 @@ impl SnsRootCanister {
         }
     }
 
+    pub async fn maybe_update_framework_canisters_memory_limit(
+        self_ref: &'static LocalKey<RefCell<Self>>,
+        management_canister_client: &impl ManagementCanisterClient,
+    ) {
+        if let Some(canisters) = self_ref.with(|state: &RefCell<SnsRootCanister>| {
```

Comment:
This if let, followed by the for loop... kind of confusing.  Can we flatten it?

Reply:
Okay, I flattened it a bit